src/service/http-server.integration.test.ts

```diff
@@ -0,0 +1,232 @@
+
+    const pixelDiff = compareImage('graph', response.body);
+    expect(pixelDiff).toBeLessThan(imageDiffThreshold);
+  });
+
+  it('should respond with the table panel screenshot', async () => {
+    const url = `${getGrafanaEndpoint(domain)}/${dashboardUid}?panelId=${panelIds.table}&render=1&from=1699333200000&to=1699344000000`;
+    const response = await request(server.app)
+      .get(
+        `/render?url=${encodeURIComponent(
+          url
+        )}&timeout=5&renderKey=${renderKey}&domain=${domain}&width=${imageWidth}&height=${imageHeight}&deviceScaleFactor=1`
+      )
+      .set('X-Auth-Token', '-');
+
+    expect(response.statusCode).toEqual(200);
+    expect(response.headers['content-type']).toEqual('image/png');
+
+    const pixelDiff = compareImage('table', response.body);
+    expect(pixelDiff).toBeLessThan(imageDiffThreshold);
+  });
+
+  it('should respond with a panel error screenshot', async () => {
+    const url = `${getGrafanaEndpoint(domain)}/${dashboardUid}?panelId=${panelIds.error}&render=1&from=1699333200000&to=1699344000000`;
+    const response = await request(server.app)
+      .get(
+        `/render?url=${encodeURIComponent(
+          url
+        )}&timeout=5&renderKey=${renderKey}&domain=${domain}&width=${imageWidth}&height=${imageHeight}&deviceScaleFactor=1`
+      )
+      .set('X-Auth-Token', '-');
+
+    expect(response.statusCode).toEqual(200);
+    expect(response.headers['content-type']).toEqual('image/png');
+
+    const pixelDiff = compareImage('error', response.body);
+    expect(pixelDiff).toBeLessThan(imageDiffThreshold);
+  });
+
+  // TODO: this doesn't work on Drone
+  // it('should timeout when a panel is too slow to load', async () => {
+  //   const url = `${getGrafanaEndpoint(domain)}/${dashboardUid}?panelId=${panelIds.slow}&render=1&from=1699333200000&to=1699344000000`;
+  //   const response = await request(server.app)
+  //     .get(
+  //       `/render?url=${encodeURIComponent(
+  //         url
+  //       )}&timeout=5&renderKey=${renderKey}&domain=localhost&width=${imageWidth}&height=${imageHeight}&deviceScaleFactor=1`
+  //     )
+  //     .set('X-Auth-Token', '-');
+
+  //   expect(response.statusCode).toEqual(200);
+  //   expect(response.headers['content-type']).toEqual('image/png');
+
+  //   const pixelDiff = compareImage('slow', response.body);
+  //   expect(pixelDiff).toBeLessThan(imageDiffThreshold);
+  // }, 30000);
+});
+
+// compareImage returns the number of different pixels between the image stored in the test file and the one from the response body.
+// It updates the stored file and returns 0 if tests are run with UPDATE_GOLDEN=true.
+// It writes the diff file to /testdata if tests are run with SAVE_DIFF=true.
+function compareImage(testName: string, responseBody: any): number {
+  const goldenFilePath = path.join(goldenFilesFolder, `${testName}.png`);
+  if (process.env['UPDATE_GOLDEN'] === 'true') {
```

Comment:
What about updating the `test-update` command to: 
```
"test-update": "cross-env UPDATE_GOLDEN=true jest -t=screenshot",
```
I'm worried that if we create a separate script, we will have a lot of duplicated code with the test. But I agree that we don't need to run all the tests so this would only run the screenshot test. We could do the same for the `test-diff` command.

Reply:
The update script should be a loop over a map with a request and writing the response to an image. I think it should be decoupled entirely of the test suite. Maybe sounds like duplicating because that is what primarily tests are, just a request and comparing the response.
The map could contain a URL (the one used to render a panel) and an image name.
However, this shouldn't be a blocker, maybe this is too much and we can move on with this PR that is a great start and refactor later if needed.